### ISSUE TITLE
Websocket failure

### ISSUE THREAD

**Issue by charlesmoore99** [heroku/webapp-runner#99]
I have a websocket (it's cribbed it from the apache websocket chat example).  

The same war file that works under a standalone Tomcat 8+ fails when used with webapp-runner..

When I attempt to use it through webapp-runner I get a 404 when attempting to access the ws: URL.

The rest of the webapp works just as it does under a standalone Tomcat.

I execute the webapp with the command line 

> java -jar target/dependency/webapp-runner.jar target/clocks.war --path /clocks

Breakpoints placed in the serverEndpoint are never hit leading me to believe that the annotation is not being discovered.

---------------
```
package clocks.chat;

import java.io.IOException;
import java.util.Set;
import java.util.concurrent.CopyOnWriteArraySet;

import javax.websocket.OnClose;
import javax.websocket.OnError;
import javax.websocket.OnMessage;
import javax.websocket.OnOpen;
import javax.websocket.Session;
import javax.websocket.server.PathParam;
import javax.websocket.server.ServerEndpoint;

import org.apache.juli.logging.Log;
import org.apache.juli.logging.LogFactory;

@ServerEndpoint(value = "/commChannel/{vewId}")
public class CommChannel {

	private static final Log log = LogFactory.getLog(CommChannel.class);

	private static final Set<CommChannel> connections = new CopyOnWriteArraySet<>();

	private Session session;

	public CommChannel() {
		log.debug("started");
	}

	@OnOpen
	public void start(Session session, @PathParam("vewId") final String vewId) {
		session.getUserProperties().put("vewId", vewId);
		this.session = session;
		connections.add(this);
	}

	@OnClose
	public void end() {
		connections.remove(this);
	}

	@OnMessage
	public void incoming(String message) {
		String vewId = (String) session.getUserProperties().get("vewId");
		String filteredMessage = String.format("%s", message.toString());
		for (Session s : session.getOpenSessions()) {
			if (s.isOpen() && vewId.equals(s.getUserProperties().get("vewId"))) {
				broadcast(vewId, filteredMessage);
			}
		}

	}

	@OnError
	public void onError(Throwable t) throws Throwable {
		log.error("Chat Error: " + t.toString(), t);
	}

	private static void broadcast(String vewId, String msg) {
		for (CommChannel client : connections) {
			try {
				synchronized (client) {
					if (client.session.getUserProperties().get("vewId").equals(vewId))
						client.session.getBasicRemote().sendText(msg);
				}
			} catch (IOException e) {
				log.debug("Chat Error: Failed to send message to client", e);
				connections.remove(client);
				try {
					client.session.close();
				} catch (IOException e1) {
				}
			}
		}
	}
}
```

**Comment by charlesmoore99** [comment]
Funny thing.  I googled the heck out of this problem yesterday before I typed up this issue, but 15 min after I his submit I find that per this [link](http://boye1024.blogspot.com/2016/01/how-to-develop-tomcat-based-websocket.html) you simply need to add the below dependency to the pom file to make it work.

```
<dependency>
	<groupId>org.apache.tomcat.embed</groupId>
	<artifactId>tomcat-embed-websocket</artifactId>
	<version>${tomcat.version}</version>
</dependency>
```	

And now my websocket works....

Should this dependency be included in the base project?
